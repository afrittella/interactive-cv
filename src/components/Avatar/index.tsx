import React from 'react';
import { IStyledBase, TCallback } from 'helpers/definitions';
import styled from 'styled-components';

interface IAvatarProps extends IStyledBase {
    imageUrl: string;
    onClick?: TCallback;
}

const Avatar: React.FunctionComponent<IAvatarProps> = ({
    imageUrl,
    onClick = () => {},
    className,
}) => {
    const handleClick = (ev: React.MouseEvent) => {
        ev.preventDefault();
        onClick();
    };

    return (
        <div
            className={className}
            style={{ backgroundImage: `url('${imageUrl}` }}
        >
            <a href="/" onClick={handleClick}>
                {' '}
            </a>
        </div>
    );
};

export default styled(React.memo(Avatar))`
    border-radius: 50%;
    width: 90px;
    height: 90px;
    background-color: ${props => props.theme.colorMedium};
    background-position: center center;
    background-size: cover;
    border: 2px solid ${props => props.theme.colorMidLight};
    margin-bottom: ${props => props.theme.sSize};

    a {
        width: 100%;
        height: 100%;
        display: block;
        border-radius: 50%;
    }
`;
