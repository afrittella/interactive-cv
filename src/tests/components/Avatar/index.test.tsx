import React from 'react';
import { render } from '@testing-library/react';
import Avatar from 'components/Avatar';
import { createBrowserHistory } from 'history';

const history = createBrowserHistory();

jest.spyOn(history, 'push');

test('renders avatar', () => {
    const { container } = render(<Avatar imageUrl="testImage" />);

    expect(container.firstChild).toMatchSnapshot();
});
