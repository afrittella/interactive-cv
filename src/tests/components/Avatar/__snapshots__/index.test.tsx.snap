// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders avatar 1`] = `
<div
  class="avatar"
>
  <a
    href=""
  >
     
  </a>
</div>
`;
